### DISCUSSION TITLE
Add Nitrogen as a Phase

### DISCUSSION THREAD

**Discussion by ruchajk2597** [PMEAL/OpenPNM#2535] [Q&A] [answered]
I am using OpenPNM to determine permeabilities through some porous structures and I want to use Nitrogen as a phase. Can I please get some guidance as to how to do this and if it is possible. I looked into the functions such as GenericPhase and Gas by name but I am not sure how this works as by traditional way it gives me errors. 

Looking forward to hearing from the team. 
Best Regards.

**Comment by jgostick** [comment, marked as answer]
If you're doing just permeability then it's pretty simple, and you may actually be over thinking it.  The permeability only needs to know the gas viscosity, so you can just create an GenericPhase object, then assign the viscosity of N2 as a constant, like:

``` python
n2 = op.phases.GenericPhase(network=pn)
n2['pore.viscosity'] = 1.8e-5  # Or whatever it is

```

Note that the StokesFlow algorithm in OpenPNM does not account for compressibility, so you don't even need molecular weight or anything.  

In the new, upcoming version of openpnm, we have more support for different fluids, but that is not really need for what you're doing here.
